php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1498 @@
+        b.zeroOrMore(ELSEIF_CLAUSE()),
+        b.optional(ELSE_CLAUSE())
+      ));
+  }
+
+  public IfStatementTree ALTERNATIVE_IF_STATEMENT() {
+    return b.<IfStatementTree>nonterminal(PHPLexicalGrammar.ALTERNATIVE_IF_STATEMENT)
+      .is(f.alternativeIfStatement(
+        b.token(PHPKeyword.IF),
+        PARENTHESIZED_EXPRESSION(),
+        b.token(PHPPunctuator.COLON),
+        b.zeroOrMore(INNER_STATEMENT()),
+        b.zeroOrMore(ALTERNATIVE_ELSEIF_CLAUSE()),
+        b.optional(ALTERNATIVE_ELSE_CLAUSE()),
+        b.token(PHPKeyword.ENDIF),
+        EOS()
+      ));
+  }
+
+  public ElseClauseTree ELSE_CLAUSE() {
+    return b.<ElseClauseTree>nonterminal(PHPLexicalGrammar.ELSE_CLAUSE)
+      .is(f.elseClause(b.token(PHPKeyword.ELSE), STATEMENT()));
+  }
+
+  public ElseifClauseTree ELSEIF_CLAUSE() {
+    return b.<ElseifClauseTree>nonterminal(PHPLexicalGrammar.ELSEIF_CLAUSE)
+      .is(f.elseifClause(
+        b.token(PHPKeyword.ELSEIF),
+        PARENTHESIZED_EXPRESSION(),
+        STATEMENT()
+      ));
+  }
+
+  public ElseClauseTree ALTERNATIVE_ELSE_CLAUSE() {
+    return b.<ElseClauseTree>nonterminal(PHPLexicalGrammar.ALTERNATIVE_ELSE_CLAUSE)
+      .is(f.alternativeElseClause(
+        b.token(PHPKeyword.ELSE),
+        b.token(PHPPunctuator.COLON),
+        b.zeroOrMore(INNER_STATEMENT())
+      ));
+  }
+
+  public ElseifClauseTree ALTERNATIVE_ELSEIF_CLAUSE() {
+    return b.<ElseifClauseTree>nonterminal(PHPLexicalGrammar.ALTERNATIVE_ELSEIF_CLAUSE)
+      .is(f.alternativeElseifClause(
+        b.token(PHPKeyword.ELSEIF),
+        PARENTHESIZED_EXPRESSION(),
+        b.token(PHPPunctuator.COLON),
+        b.zeroOrMore(INNER_STATEMENT())
+      ));
+  }
+
+  public ForStatementTree FOR_STATEMENT() {
+    return b.<ForStatementTree>nonterminal(PHPLexicalGrammar.FOR_STATEMENT)
+      .is(b.firstOf(
+          f.forStatement(
+            FOR_STATEMENT_HEADER(),
+            STATEMENT()
+          ),
+          f.forStatementAlternative(
+            FOR_STATEMENT_HEADER(),
+            b.token(PHPPunctuator.COLON),
+            b.zeroOrMore(INNER_STATEMENT()),
+            b.token(PHPKeyword.ENDFOR),
+            EOS()
+          ))
+      );
+  }
+
+  public ForStatementHeader FOR_STATEMENT_HEADER() {
+    return b.<ForStatementHeader>nonterminal()
+      .is(f.forStatementHeader(
+        b.token(PHPKeyword.FOR), b.token(PHPPunctuator.LPARENTHESIS),
+        b.optional(FOR_EXPR()),
+        b.token(PHPPunctuator.SEMICOLON),
+        b.optional(FOR_EXPR()),
+        b.token(PHPPunctuator.SEMICOLON),
+        b.optional(FOR_EXPR()),
+        b.token(PHPPunctuator.RPARENTHESIS)
+      ));
+  }
+
+  public SeparatedList<ExpressionTree> FOR_EXPR() {
+    return b.<SeparatedList<ExpressionTree>>nonterminal(PHPLexicalGrammar.FOR_EXPR)
+      .is(f.forExpr(
+        EXPRESSION(),
+        b.zeroOrMore(f.newTuple12(b.token(PHPPunctuator.COMMA), EXPRESSION()))
+      ));
+  }
+
+  public ForEachStatementTree FOREACH_STATEMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)